[SDESK-7497] Small adjustments for `superdesk-planning` behave tests

eos87 opened the pull request:
Small changes required for `superdesk-planning` behave tests to pass. Some of them are just for debugging.

[SDESK-7497]

[SDESK-7497]: https://sofab.atlassian.net/browse/SDESK-7497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

eos87 committed:
Small adjustments for superdesk-planning behave tests

SDESK-7497

eos87 commented on apps/publish/published_item.py:
```diff
@@ -137,6 +137,17 @@ class PublishedItemService(BaseService, HighlightsSearchMixin):
+        FIXME: Temporary workaround until we find out why _id and guid are not in sync
+        when running tests from `superdesk-planning`. Without this, many tests fill fail.
+        """
+        item = self.find_one(req=None, guid=id)
+        if item:
+            id = item[ID_FIELD]
```
I could not find the root of this problem unfortunately. I just added this workaround with the full intention of finding why are the `_id` and `guid` out of sync. So far what I've found is that when the `published` item is created, the values are fine, but apparently there is some action happening in between that is modifying the `_id` with an `ObjectId` value.

MarkLark86 replied:
@eos87 The `_id` field on the `published` collection uses an ObjectId. There could be multiple entries for the one item (multiple versions). What we use instead is the `item_id` field, as that is what matches the `_id` or `guid` from the `archive` collection (and other content based collections).

Where specifically in the Planning module is this causing an issue?

eos87 replied:
Removed as we spoke earlier today.

eos87 committed:
Allow empty async resources

SDESK-7497

eos87 committed:
Add missing context block

eos87 committed:
Rollback workaround as it won't be needed

 Some newer implementation of the publishing engine will most likely solve the problem

MarkLark86 commented on apps/publish/content/common.py:
```diff
@@ -242,9 +242,12 @@ def update(self, id, updates, original):
                 published_service = get_resource_service(PUBLISHED)
                 assert published_service is not None
+
+                # TODO-ASYNC: superdesk-planning tests break because of difference between `_id` and `guid`
```
These 2 comments aren't needed, you can see the changes in my PR here:

https://github.com/superdesk/superdesk-core/blob/32fb1ba9c9a798b01a431d153b533983675443ff/apps/publish/content/common.py#L295-L308

MarkLark86 reviewed (approved):
Looks good

eos87 committed:
Fix mypy error and remove comment
